crates/driver/src/boundary/mempool.rs

```diff
@@ -49,6 +50,16 @@ pub enum Kind {
     },
 }
 
+impl Kind {
+    /// for instrumentization purposes
+    pub fn format_variant(&self) -> String {
```

Comment:
Probably makes sense to return a &'static str to avoid allocating over and over.

Reply:
Indeed. Fixed.